x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractBlock.java

```diff
@@ -100,43 +100,6 @@ public boolean isReleased() {
         return hasReferences() == false;
     }
 
-    @Override
-    public final void incRef() {
-        if (isReleased()) {
-            throw new IllegalStateException("can't increase refCount on already released block [" + this + "]");
-        }
-        references++;
-    }
-
-    @Override
-    public final boolean tryIncRef() {
-        if (isReleased()) {
-            return false;
-        }
-        references++;
-        return true;
-    }
-
-    @Override
-    public final boolean decRef() {
-        if (isReleased()) {
```

Comment:
If we use `AbstractRefCounted`, we'll have to update some use of vectors, which sometimes can get released without us decrementing the refcount all the way, but still releasing their containing blocks. Currently, blocks generated from `X-VectorBlock` override `isReleased` to account for this.